lib/evse_security/evse_security.cpp

```diff
@@ -1086,6 +1086,47 @@ GetCertificateSignRequestResult EvseSecurity::generate_certificate_signing_reque
     return generate_certificate_signing_request(certificate_type, country, organization, common, false);
 }
 
+GetCertificateFullInfoResult EvseSecurity::get_all_valid_certificates_info(LeafCertificateType certificate_type,
+                                                                           EncodingFormat encoding, bool include_ocsp) {
+    std::lock_guard<std::mutex> guard(EvseSecurity::security_mutex);
+
+    GetCertificateFullInfoResult result =
+        get_full_leaf_certificate_info_internal(certificate_type, encoding, include_ocsp, true, true);
+
+    // If we failed, simply return the result
+    if (result.status != GetCertificateInfoStatus::Accepted) {
+        return result;
+    }
+
+    GetCertificateFullInfoResult filtered_results;
+    filtered_results.status = result.status;
+
+    // Filter the certificates to return only the ones that have a unique
+    // root, and from those that have a unique root, return only the newest
```

Comment:
I've discussed that internally and there is no need to support cross-signed certificates at the moment. Has been some change in the meantime that I am not aware of?

Reply:
My suggestion is to add comments so it is clear that:
- roots that are not self signed are not supported
- cross signed certificates (where a leaf has multiple roots) are not supported

Given we have no control over the CA that issues certificates there is a risk that a CA changes something that could prevent a charger for communicating with the CSMS (or cause issues with 15118).

i.e. it just needs to be clear that these are not currently supported (perhaps adding an issue to github would be a good idea)

Reply:
Sounds like a plan.

For the non-self signed roots, the provided documentation is not completely clear, because it fails to specify an example (unless i've missed it). However if the root is not self-signed, then it is placed in the hierarchy as an orphan. That does not imply that it is not usable, the partial hierarchy should still be build, and if it doesn't (again, a real-world example would be great) it is a very easy fix.